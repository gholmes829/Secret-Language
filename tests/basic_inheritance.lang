# inheritance

class Dinosaur {
    void talk() {
        print('RROOOOAAAAARRRR!!!')
    }
    void feast() {
        print('MUUUNCHHHH')
    }
}

class Chicken(Dinosaur) {
    void feast() {
        print('nibble...')
    }
}

num main() {
    let dino = Dinosaur()
    print(dino)
    dino.talk()
    dino.feast()

    let chick = Chicken()
    print(chick)
    chick.talk()
    chick.feast()

    return 0
}

main()