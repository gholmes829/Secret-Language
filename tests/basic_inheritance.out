<Internal Instance of "Dinosaur">
RROOOOAAAAARRRR!!!
MUUUNCHHHH
<Internal Instance of "Chicken">
RROOOOAAAAARRRR!!!
nibble...
